frontend/packages/dev-console/src/components/pipelines/pipeline-form/SecretForm.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikValues, useFormikContext } from 'formik';
+import { ActionGroup, Button, ButtonVariant, TextInputTypes } from '@patternfly/react-core';
+import { CheckIcon, CloseIcon } from '@patternfly/react-icons';
+import { ButtonBar } from '@console/internal/components/utils';
+import {
+  SecretType,
+  BasicAuthSubform,
+  SSHAuthSubform,
+  CreateConfigSubform,
+} from '@console/internal/components/secrets/create-secret';
+import { DropdownField, InputField } from '@console/shared';
+import './SecretForm.scss';
+
+const authTypes = {
+  [SecretType.dockerconfigjson]: 'Image Registry Credentials',
+  [SecretType.basicAuth]: 'Basic Authentication',
+  [SecretType.sshAuth]: 'SSH Key',
+};
+
+const renderSecretForm = (
+  type: string,
+  stringData: {
+    [key: string]: any;
+  },
+  onDataChanged: (event: Event) => void,
+) => {
+  switch (type) {
+    case SecretType.basicAuth:
+      return <BasicAuthSubform onChange={onDataChanged} stringData={stringData} />;
+    case SecretType.sshAuth:
+      return <SSHAuthSubform onChange={onDataChanged} stringData={stringData} />;
+    case SecretType.dockerconfigjson:
+      return <CreateConfigSubform onChange={onDataChanged} stringData={stringData} />;
+    default:
+      return null;
+  }
+};
+
+const SecretForm: React.FC<FormikValues> = ({
+  handleSubmit,
+  handleReset,
+  status,
+  isSubmitting,
+}) => {
+  const { values, setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
+  const stringData = {};
+
+  const onDataChanged = (event) => {
+    values.type === SecretType.dockerconfigjson
+      ? setFieldValue('formData', _.mapValues({ '.dockerconfigjson': event }, JSON.stringify))
```

Comment:
You're stringifying an Event object? That doesn't sound like a good idea. What does these Event objects contain?

Reply:
For ```Image Registry Credentials``` secret it accepts stringify data. Event objects contain form data. eg: username and password.